ash/src/vk/extensions.rs

```diff
@@ -17672,6 +17672,20 @@ impl ExtPipelinePropertiesFn {
     pub const SPEC_VERSION: u32 = 1u32;
 }
 #[allow(non_camel_case_types)]
+#[doc = "Implemented for all types that can be passed as argument to `p_pipeline_properties` in [`PFN_vkGetPipelinePropertiesEXT`]"]
+pub unsafe trait PFN_vkGetPipelinePropertiesEXT_param_p_pipeline_properties_validstructs {
+    unsafe fn as_base_in_struct(&self) -> *const BaseInStructure {
+        <*const _>::cast(self)
+    }
+    unsafe fn as_base_out_struct(&mut self) -> *mut BaseOutStructure {
+        <*mut _>::cast(self)
+    }
+}
+unsafe impl PFN_vkGetPipelinePropertiesEXT_param_p_pipeline_properties_validstructs
+    for PipelinePropertiesIdentifierEXT<'_>
```

Comment:
When do we expect more impls to arise? If there's only the one, this seems perhaps overkill.

Reply:
Not sure, this is still the only param with `validstructs`, perhaps @oddhack is able to comment?

We could stick to `vk::BaseOutStructure` and a doc-comment in the `PipelineProperties` wrapper (effectively dropping the second commit from this PR) but it might fall out of date. Similar if we hand-roll this trait: rather generate as much as possible from the underlying `vk.xml` spec.

Reply:
I can't say this will never happen again but had I noticed it early enough I would have argued vigorously against this type punning (vs. passing a real VkBaseOutStructure or other no-op structure, and the actual structure in the pNext chain, say).

Reply:
Is this a pattern you don't want to support going forward? If so you might still introduce such a no-op structure and leverage existing `structextends` machinery for it, and somehow deprecate this attribute notion leaving it only for this `p_pipeline_properties` param (extended to also accept a no-op structure with the current `VkPipelinePropertiesIdentifierEXT` in `pNext`)?

Reply:
We'd have to create another extension to allow this, so I doubt there would be appetite for it in the WG. Avoiding the pattern in extensions introducing new APIs is probably doable.

Reply:
All the machinery is there to support it except that drivers are now forced to read the pNext chain?

Fine to keep using this construct to be honest, we just added support for it.